messages-service.js

```diff
@@ -28,13 +31,59 @@ function factory($http, config) {
         .then(function(results) {
           var messages = results.data;
           service.unreadCount = messages.filter(function(message) {
-            return !message.meta.read;
+            return !message.meta.read && !message.meta.archived;
           }).length;
           return new Promise(function(resolve, reject) {
             resolve(results);
           });
         });
   };
+
+  service.delete = function(message) {
+    return Promise.resolve($http.delete(messagesEndpoint + '/' + message.id));
+  };
+
+  service.deleteBatch = function(messages) {
+    var request = {};
+    request.messages = messages;
+
+    var dataString = JSON.stringify(request);
```

Comment:
Could just move this `JSON.stringify` down and remove the `dataString` var for brevity.